data/store/mongo/mongo_alerts.go

```diff
@@ -0,0 +1,55 @@
+package mongo
+
+import (
+	"context"
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/tidepool-org/platform/alerts"
+	"github.com/tidepool-org/platform/store/structured/mongo"
+)
+
+// alertsRepo implements alerts.Repository, writing data to a MongoDB collection.
+type alertsRepo mongo.Repository
+
+// Upsert will create or update the given Config.
+func (r *alertsRepo) Upsert(ctx context.Context, conf *alerts.Config) error {
+	opts := options.Update().SetUpsert(true)
+	doc := NewAlertsConfigDocument(conf)
+	filter := bson.M{"_id": doc.ID}
+	_, err := r.UpdateOne(ctx, filter, bson.M{"$set": doc}, opts)
+	if err != nil {
+		return fmt.Errorf("upserting AlertsConfig: %w", err)
+	}
+	return nil
+}
+
+// Delete will delete the given Config.
+func (r *alertsRepo) Delete(ctx context.Context, conf *alerts.Config) error {
+	filter := bson.M{"_id": AlertsID(conf)}
+	_, err := r.DeleteMany(ctx, filter, nil)
+	if err != nil {
+		return fmt.Errorf("upserting AlertsConfig: %w", err)
+	}
+	return nil
+}
+
+// AlertsConfigDocument wraps alerts.Config to provide an ID for mongodb.
+type AlertsConfigDocument struct {
+	ID             string `bson:"_id"`
```

Comment:
You're overriding the autogenerated ObjectId `_id` with a string identifier of the concatenated "ownerId:invitorId". There's a unique constraint on that index that cannot be removed. This implies the alert configuration is unique for the user id tuple. Is there a possibly down that we may allow multiple alerts/alert configurations?

Reply:
You tell me? Aside from a "never say never" situation, I don't know what that would be...

Is it bad form in mongo to override the generated IDs? I can move away from that... I was unsure about it in the first place, but I _do_ want to ensure that, at least for now, there is only a single alerts config for any given combination of owner and invitor.